packages/angular_devkit/build_angular/src/angular-cli-files/utilities/check-port.ts

```diff
@@ -5,26 +5,32 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+import * as net from 'net';
 
-import { Observable } from 'rxjs';
-const portfinder = require('portfinder');
+export function checkPort(port: number, host: string, basePort = 49152): Promise<number> {
+  return new Promise<number>((resolve, reject) => {
+    function _getPort(portNumber: number) {
+      if (portNumber > 65535) {
+        reject(new Error(`There is no port available.`));
+      }
 
+      const server = net.createServer();
 
-export function checkPort(port: number, host: string, basePort = 49152): Observable<number> {
-  return new Observable(obs => {
-    portfinder.basePort = basePort;
-    // tslint:disable:no-any
-    portfinder.getPort({ port, host }, (err: any, foundPort: number) => {
-      if (err) {
-        obs.error(err);
-      } else if (port !== foundPort && port !== 0) {
-        // If the port isn't available and we weren't looking for any port, throw error.
-        obs.error(`Port ${port} is already in use. Use '--port' to specify a different port.`);
-      } else {
-        // Otherwise, our found port is good.
-        obs.next(foundPort);
-        obs.complete();
-      }
-    });
+      server.once('error', err => {
+        if (port !== 0 && err.code === 'EADDRINUSE') {
+          // If the port isn't available and we weren't looking for any port, throw error.
+          reject(new Error(`Port ${port} is already in use. Use '--port' to specify a different port.`));
```

Comment:
You don't want to fallback to calling `_getPort()` here. You should return (or wrap the `_getPort()` call in an `else` clause.

Reply:
What about something similar to the following? It only checks the error code once and is more explicit that port === 0 has special handling.
```  if (err.code !== 'EADDRINUSE') {
    reject(err);
  } else if (port === 0) {
    _getPort(portNumber + 1);
  } else {
    reject(new Error(`Port ${port} is already in use. Use '--port' to specify a different port.`));
  }
```

The zero handling logic can also be removed in an upcoming major release.  The OS knows how to handle the value.  The check can be completely skipped in that situation as well.

Reply:
Beautiful!!
Squash the commits too?

Reply:
Yes please.